Firestore/core/src/util/byte_stream_apple.mm

```diff
@@ -95,8 +83,23 @@
   return static_cast<int32_t>(read);
 }
 
-StreamReadResult ByteStreamApple::EofResult() {
-  return StreamReadResult(std::move(buffer_), true);
+StreamReadResult ByteStreamApple::ConsumeBuffer() {
```

Comment:
To be clear, I suggested passing the value `eof` as a boolean parameter -- I presume it would avoid the need to do the peeking. Is there a reason peeking is preferable?

Reply:
Yes. NSInputStream seems to return `eof:false` with `Read(0)` on `""`, while `istream` behaves otherwise.

Essentially, the behaviour btw the two are not consistent, explicit peeking makes the them consistent.